src/main/java/com/xceptance/xlt/engine/htmlunit/okhttp3/CookieJarImpl.java

```diff
@@ -105,8 +82,11 @@ private static boolean matchesUri(final Cookie htmlUnitCookie, final HttpUrl url
      */
     private static Cookie toHtmlUnitCookie(final okhttp3.Cookie okHttpCookie)
     {
-        return new Cookie(okHttpCookie.domain(), okHttpCookie.name(), okHttpCookie.value(), okHttpCookie.path(),
-                          new Date(okHttpCookie.expiresAt()), okHttpCookie.secure(), okHttpCookie.httpOnly());
+        // add a leading dot if the domain is a "wildcard" domain -> ".example.com"
+        final String domain = okHttpCookie.hostOnly() ? okHttpCookie.domain() : "." + okHttpCookie.domain();
```

Comment:
Are OkHTTP Cookies guaranteed to not start with a `.` character? If not, we'll have to use something like `StringUtils.prependIfMissing` in order to avoid cases when the domain of domain cookie already starts with a `.`.

Reply:
Yep, OkHttp cookie domains never start with a `.`. To tell wildcard domains from host domains, the host-only flag is set to true/false accordingly.